app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,49 @@
+class CommentsController < ApplicationController
+  before_action :authenticate_to_submit, only: :new
+  before_action :authenticate_user!, only: :create
+  before_action :parent, only: :create
```

Comment:
this is completely unnecessary if `parent` was referenced in the `create` action.